code/service/src/main/scala/com/namely/chiefofstate/readside/ReadSideHandler.scala

```diff
@@ -103,27 +106,55 @@ private[readside] class ReadSideHandlerImpl(
   }
 }
 
-/**
- * Processes events read from the Journal
- *
- * @param event          the actual event
- * @param eventTag       the event tag
- * @param resultingState the resulting state of the applied event
- * @param meta           the additional meta data
- * @return an eventual HandleReadSideResponse
- */
 private[readside] trait ReadSideHandler {
 
+  implicit val ec: ExecutionContext
+
   /**
-   * handles a read side message
+   * Processes events read from the Journal.
+   * Exponentially backoff with a 10& gain modifier until it reaches the upper threshold of maxBackoffSeconds.
    *
-   * @param event
-   * @param eventTag
-   * @param resultingState
-   * @param meta
-   * @return
+   * @param event               the actual event
+   * @param eventTag            the event tag
+   * @param resultingState      the resulting state of the applied event
+   * @param meta                the additional meta data
+   * @param policy              the retry policy. If provided, overrides the default backoff policy. Testing method
+   * @param minBackoffSeconds   the minimum number of seconds to backoff
+   * @param maxBackoffSeconds   the maximum number of seconds to backoff
+   * @return                    Future[Boolean] for success
    */
   def processEvent(
+      event: com.google.protobuf.any.Any,
+      eventTag: String,
+      resultingState: com.google.protobuf.any.Any,
+      meta: MetaData,
+      policy: Option[retry.Policy] = Some(retry.Directly()),
+      minBackoffSeconds: Long = 1L,
+      maxBackoffSeconds: Long = 30L): Future[Boolean] = {
+
+    implicit val success: retry.Success[Boolean] = retry.Success(x => x)
+
+    val finalPolicy: retry.Policy =
```

Comment:
this implementation only makes sense for gRPC. Just put it in the `impl` class.

FYI - this trait was only meant to make testing easier, not really for implementing many of these.

Reply:
The policy is not a grpc concept though. It is a retry concept.